Only treat first buffer as .glb-embedded

zeux opened the pull request:
Instead of treating all buffers with non-existent URI as being embedded
into GLB file, we only treat the first buffer as embedded - we now emit
a validation error if the first buffer in GLB file has a URI though.

Fixes #121.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/KhronosGroup/glTF-Validator?pullRequest=122) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/KhronosGroup/glTF-Validator?pullRequest=122) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/KhronosGroup/glTF-Validator?pullRequest=122) it.</sub>

zeux commented:
This change breaks "00_load_buffers_test.dart: Load Buffers Non first buffer refers to GLB" but I'm not sure this test is valid?

The spec says:

> glTF asset could use GLB file container to pack all resources into one file. glTF Buffer referring to GLB-stored BIN chunk, must have buffer.uri property undefined, and it must be the first element of buffers array; byte length of BIN chunk could be up to 3 bytes bigger than JSON-defined 
buffer.byteLength to satisfy GLB padding requirements.

I think this says "the first buffer in a .glb file that has buffer.uri = null needs to refer to BIN chunk" - ok, that's what my change is testing.
I think this does *not* say "the first buffer in a .glb file *must* refer to BIN chunk" - what if no buffers refer to BIN chunk but the chunk is still there? What if no buffers refer to BIN chunk, buffer 1 happens to have uri = null, but the BIN chunk is still there?

It seems to me that while it's not obvious immediately, non_first_buffer.glb is actually valid (it has two buffers, one has a URI, one is just empty/dummy) but I'm not sure if my treatment of the spec is accurate.

lexaknyazev commented:
> I think this does not say "the first buffer in a .glb file must refer to BIN chunk" - what if no buffers refer to BIN chunk but the chunk is still there? What if no buffers refer to BIN chunk, buffer 1 happens to have `uri` = null, but the BIN chunk is still there?

These are all good questions that fall into "under-defined" spec territory. Before doing anything here, let's clarify the spec first. Could you please open a new spec issue (referring the original use-case of the proposed extension) that would clearly list all questionable corner cases? Thanks very much.

zeux commented:
I've updated the code to match the spec clarification, and updated the tests to match. CI fails due to dartanalyzer complaining about the code which seems unrelated...

zeux commented:
@lexaknyazev this should be ready for review since the spec questions have been cleared up

lexaknyazev commented:
This issue was fixed internally as a part of `2.0.0-dev.3.0` release.
